extensions/PhabBugz/lib/Util.pm

```diff
@@ -0,0 +1,310 @@
+
+    my $data = {
+        transactions => [
+            {
+                type  => 'view',
+                value => $policy_phid
+            },
+            {
+                type  => 'edit',
+                value => $policy_phid
+            }
+        ],
+        objectIdentifier => $revision_phid
+    };
+
+    if (@$subscribers) {
+        push(
+            @{ $data->{transactions} },
+            {
+                type  => 'subscribers.add',
+                value => $subscribers
+            }
+        );
+    }
+
+    return request( 'differential.revision.edit', $data );
+}
+
+sub get_project_phid {
+    my $project = shift;
+
+    my $data = {
+        queryKey    => 'all',
+        constraints => {
+            name => $project
+        }
+    };
+
+    my $result = request( 'project.search', $data );
+    return undef
+      unless ( exists $result->{result}{data} && @{ $result->{result}{data} } );
+
+    return $result->{result}{data}[0]{phid};
+}
+
+sub create_project {
+    my ( $project, $description, $members ) = @_;
+
+    my $data = {
+        transactions => [
+            { type => 'name',        value => $project },
+            { type => 'description', value => $description },
+            { type => 'edit',        value => 'admin' },
+            { type => 'join',        value => 'admin' },
+            { type => 'view',        value => 'admin' },
+            { type => 'icon',        value => 'group' },
+            { type => 'color',       value => 'red' }
+        ]
+    };
+
+    my $result = request( 'project.edit', $data );
+    return $result->{result}{object}{phid};
+}
+
+sub set_project_members {
+    my ( $project_id, $phab_user_ids ) = @_;
+
+    my $data = {
+        objectIdentifier => $project_id,
+        transactions => [ { type => 'members.set', value => $phab_user_ids } ]
+    };
+
+    my $result = request( 'project.edit', $data );
+    return $result->{result}{object}{phid};
+}
+
+sub get_members_by_bmo_id {
+    my $users = shift;
+
+    my $data = { accountids => [ map { $_->id } @$users ] };
+
+    my $result = request( 'bmoexternalaccount.search', $data );
+    return [] if ( !$result->{result} );
+
+    my @phab_ids;
+    foreach my $user ( @{ $result->{result} } ) {
+        push( @phab_ids, $user->{phid} );
+    }
+
+    return \@phab_ids;
+}
+
+sub request {
+    my ( $method, $data ) = @_;
+    my $request_cache = Bugzilla->request_cache;
+    my $params        = Bugzilla->params;
+
+    my $ua = $request_cache->{phabricato_ua};
+    unless ($ua) {
+        $ua = $request_cache->{phabricator_ua} =
+          LWP::UserAgent->new( timeout => 10 );
+        if ( $params->{proxy_url} ) {
+            $ua->proxy( 'https', $params->{proxy_url} );
+        }
+        $ua->default_header(
+            'Content-Type' => 'application/x-www-form-urlencoded' );
+    }
+
+    my $phab_api_key = $params->{phabricator_api_key};
+    ThrowUserError('invalid_phabricator_api_key') unless $phab_api_key;
```

Comment:
this is fine, but in this case "or Throw" is actually fine too, because the main purpose of the code is to set the variable. Just an opinion though.